js/home.js

```diff
@@ -175,3 +298,148 @@ function addFavoriteButtonEvent () {
     })
   })
 }
+
+function showModal (id, event) {
```

Comment:
Attempt to use this showModal method instead. I got it working by making these changes on top of yours, so hopefully it will for you as well.

```javascript
function showModal (id, event) {

  let pokemonMap = null

  pokemons.pokemon.forEach(species => {
    let pokemon = species.pokemon.find(pokemonById => {
      console.log(pokemonById, id)
      return pokemonById['id'] === id
    })

    if (pokemon) pokemonMap = {species, pokemon}
  })

  if (!pokemonMap) {
    console.error("No Pokemon Found to Display Detail")
    return
  }

  let $detailModal = $(detailModal)

  $detailModal.find('.modal-title').text(pokemonMap.species.name)
  $detailModal.find('.modal-body').html(detailModalBody(pokemonMap.pokemon))
  $detailModal.modal('show')
}
```